tmux/tmux.go

```diff
@@ -7,11 +7,82 @@ import (
 	"os"
 	"os/exec"
 	"strings"
+	"sync"
 
 	"github.com/joshmedeski/sesh/config"
 	"github.com/joshmedeski/sesh/dir"
 )
 
+var (
+	command *Command
+	once    sync.Once
+)
+
+func init() {
+	once.Do(func() {
+		var err error
+		command, err = NewCommand()
+		if err != nil {
+			log.Fatal(err)
+		}
+	})
+}
+
+type Error struct{ msg string }
+
+func (e Error) Error() string { return e.msg }
+
+var ErrNotRunning = Error{"no server running"}
+
+func executeCommand(command string, args []string) (string, error) {
+	var stdout, stderr bytes.Buffer
+	cmd := exec.Command(command, args...)
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	if err := cmd.Start(); err != nil {
+		return "", err
+	}
+
+	if err := cmd.Wait(); err != nil {
+		if strings.Contains(stderr.String(), "no server running on") {
+			return "", ErrNotRunning
+		}
+
+		return "", err
+	}
+
+	out := strings.TrimSpace(stdout.String())
+	if strings.Contains(out, "no server running on") {
+		return "", ErrNotRunning
+	}
+
+	return out, nil
+}
+
+type Command struct {
+	cliPath  string
+	execFunc func(string, []string) (string, error)
+}
+
+func NewCommand() (c *Command, err error) {
```

Comment:
Can all this generic command code be moved to a package that zoxide and tmux can share?

Reply:
Some of the core command execution code could be extracted to a common package, but that would mean loosing some of the isolation / safety we get from the use of private variables / methods. 

I'd typically lean towards the duplication of a small amount of code for the isolation and flexibility it gives, but if you have a strong prefence here I can extract the command execution parts.